src/arrays/creating_arrays/hard/q2/MainTest.java

```diff
@@ -0,0 +1,105 @@
+package arrays.creating_arrays.hard.q2;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.utils.ArrayUtil;
+import global.variables.Clause;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import global.variables.wrappers.Optional;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    // Java
+
+    final static int num = 5;
+
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = ArrayUtil.arrayToInput(ArrayUtil.generateRandomArray(5, 200, num));
+        return clauseBuilder();
+    }
+
+    public void runMain() {
+        RevdArray.main(new String[0]);
+    }
+
+    static Stream<Arguments> reverseInputProvider() {
+        return Stream.of(
+                Arguments.of(0, 0),
+                Arguments.of(1, 1),
+                Arguments.of(1000, 1),
+                Arguments.of(24, 42),
+                Arguments.of(3109, 9013),
+                Arguments.of(9, 9),
+                Arguments.of(73821, 12837),
+                Arguments.of(73, 37)
+        );
+    }
+
+    static Stream<int[]> mainMethodInputProvider() {
+        return Stream.of(
+                ArrayUtil.generateRandomArray(5, 200, num),
+                ArrayUtil.generateRandomArray(5, 200, num),
+                ArrayUtil.generateRandomArray(5, 200, num),
+                ArrayUtil.generateRandomArray(5, 200, num)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("reverseInputProvider")
+    void correctReverseMethod(int in, int ans) throws Throwable {
+        Object[][] arguments = {
+                {in, int.class}
+        };
+        MethodTest m = new MethodTest(RevdArray.class, "reverse", arguments);
+        Object output = m.callMethod();
+        CustomAssertions._assertEquals(ans, output, "Your reverse method does not correctly reverse the given number.");
+    }
+
+    @ParameterizedTest
+    @MethodSource("mainMethodInputProvider")
+    void printsOutputCorrectly(int[] in) {
+        runWithInput(ArrayUtil.arrayToInput(in));
+
+        for(int x = 0; x < in.length; x++){
+            assertEquals(solution(in[x]), Integer.parseInt(getItemByName("num" + x)), "Your program does not correctly reverse array element" + x);
```

Comment:
I don't think you need to specify the array element, I think you can make the message a bit more generic. 
```suggestion
            assertEquals(solution(in[x]), Integer.parseInt(getItemByName("num" + x)), "Your program does not correctly reverse each element in the array." + x);
```